libraries/classes/ThemeGenerator.php

```
@@ -0,0 +1,216 @@
+<?php
+/**
+ * Holds the PhpMyAdmin\ThemeGenerator class
+ *
+ * @package PhpMyAdmin
+ */
+declare(strict_types=1);
+
+namespace PhpMyAdmin;
+
+use PhpMyAdmin\Template;
+
+/**
+ * Set of functions for Automated Theme Generator in phpMyAdmin
+ *
+ * @package PhpMyAdmin
+ */
+class ThemeGenerator
+{
+    /** @var Template */
+    protected $template;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->template = new Template();
+    }
+
+    /**
+     * Colour Picker HTML file
+     *
+     */
+    public function colorPicker(): string
+    {
+        $output = $this->template->render('theme_generator/color_picker');
+        return $output;
+    }
+
+    /**
+     * Colour Palette HTML file
+     *
+     */
+    public function colorPalette(): string
+    {
+        $output = $this->template->render('theme_generator/color_palette');
+        return $output;
+    }
+
+    /**
+     * Preview and Save HTML file
+     *
+     */
+    public function previewAndSave(): string
+    {
+        $output = $this->template->render('theme_generator/preview_and_save');
+        return $output;
+    }
+
+    /**
+     * Table preview HTML file
+     *
+     */
+    public function tablePreview(): string
+    {
+        $output = $this->template->render('theme_generator/table_preview');
+        return $output;
+    }
+
+    /**
+     * Recursion copy function from one directory to another
+     *
+     * @param string $src Source path of the directory
+     *
+     * @param string $dst Destination path of the directory
+     *
+     */
+    public function recurse_copy(string $src, string $dst): void
+    {
+        $dir = opendir($src);
+        @mkdir($dst);
```

All the ignore errors and warnings are not the best practice, if you can avoid them please do so